src/preset.ts

```diff
@@ -931,6 +935,11 @@ async function expandConfigurePresetHelper(folder: string, preset: ConfigurePres
             // The env variables for the supported compilers are the same.
             const compilerName: string | undefined = util.isSupportedCompiler(cxxCompiler) || util.isSupportedCompiler(cCompiler);
             if (compilerName) {
+                for (const key in preset.environment) {
+                    if (preset.environment[key]) {
+                        preset.environment[key] = await expandString(preset.environment[key]!, expansionOpts);
```

Comment:
I see that we also expand the environment in existing code on line 689. Is it safe to potentially expand an environment variable twice? Consider this case:

```
${dollar}{sourceDir}
```

The correct expansion is literally `${sourceDir}`, but I'm concerned that with this change it would end up expanding to the source directory itself.

Reply:
That's a good point. Maybe we can keep a copy of the unexpanded environment variable so that we can reset it after using it for `where.exe`

Reply:
@benmcmorran I believe I fixed the issue by expanding `preset.environment` into `expandedPreset.environment` instead overwriting `preset.environment`. There are instances of this elsewhere in that file. I also had to include `name` as a parameter, but I made sure that all the references to that function includes `name` now.